peps/pep-0757.rst

```diff
@@ -263,15 +263,25 @@ Code::
             const PyLongLayout* layout = PyLong_GetNativeLayout();
             static PyLong_DigitArray long_export;
 
-            PyLong_AsDigitArray(obj, &long_export);
-            mpz_import(z, long_export.ndigits, layout->endian,
-                       layout->digit_size, layout->digits_order,
-                       layout->digit_size*8 - layout->bits_per_digit,
-                       long_export.digits);
+            PyLong_Export(obj, &long_export);
+            if (long_export.digits) {
+                mpz_import(z, long_export.ndigits, layout->digits_order,
+                           layout->digit_size, layout->endian,
+                           layout->digit_size*8 - layout->bits_per_digit,
+                           long_export.digits);
+            }
+            else {
+                if (long_export.negative) {
+                    long_export.value = -long_export.value;
+                }
+                mpz_import(z, 1, -1, sizeof(int64_t), 0, 0,
```

Comment:
Is it faster to check if *value* is in [INT_MIN; INT_MAX] to call `mpz_set_si()` instead? (otherwise, call mpz_import)